core/src/main/scala/org/apache/spark/executor/ShuffleReadMetrics.scala

```diff
@@ -119,5 +119,25 @@ class ShuffleReadMetrics private[spark] () extends Serializable {
       _recordsRead.add(metric.recordsRead)
     }
   }
+}
 
+/**
+ * A temporary shuffle read metrics holder that is used to collect shuffle read metrics for each
+ * shuffle dependency, and all temporary metrics will be merged into the [[ShuffleReadMetrics]] at
+ * last.
+ */
+private[spark] class TempShuffleReadMetrics {
+  var remoteBlocksFetched = 0
```

Comment:
this code path might be performance sensitive, so while you are at it, maybe change all of these to `private[this]`, and add an accessor for each, e.g.

```scala
private[var] var _remoteBlocksFetched = 0L
def remoteBlocksFetched: Long = _remoteBlocksFetched
def incRemoteBlocksFetched(v: Int): Unit = _remoteBlocksFetched += v
```

btw they should be long type right?